### ISSUE TITLE
pnpm link -g fails to find package.json

### ISSUE THREAD

**Issue by Clashsoft** [pnpm/pnpm#2316]
### Code to reproduce the issue:

```sh
$ pnpm i -g @angular/cli
$ ng
zsh: command not found: ng
$ pnpm link -g @angular/cli
ERROR No package.json (or package.yaml, or package.json5) was found in "/usr/local/Cellar/node/13.7.0/pnpm-global/node_modules/@angular/cli".
```

`pnpm-debug.log`:

```json
{
  "0 debug pnpm:scope": {
    "selected": 1
  },
  "1 debug pnpm": {
    "isCaseSensitive": false,
    "store": "/Users/me/.pnpm-store/2"
  },
  "2 error pnpm": {
    "message": {
      "code": "ERR_PNPM_NO_IMPORTER_MANIFEST_FOUND"
    },
    "err": {
      "name": "Error",
      "message": "No package.json (or package.yaml, or package.json5) was found in \"/usr/local/Cellar/node/13.7.0/pnpm-global/node_modules/@angular/cli\".",
      "code": "ERR_PNPM_NO_IMPORTER_MANIFEST_FOUND",
      "stack": "Error: No package.json (or package.yaml, or package.json5) was found in \"/usr/local/Cellar/node/13.7.0/pnpm-global/node_modules/@angular/cli\".\n    at readProjectManifest (/usr/local/Cellar/pnpm/4.7.1/libexec/lib/node_modules/pnpm/lib/node_modules/@pnpm/read-project-manifest/lib/index.js:20:11)\n    at async Object.readProjectManifestOnly (/usr/local/Cellar/pnpm/4.7.1/libexec/lib/node_modules/pnpm/lib/node_modules/@pnpm/read-project-manifest/lib/index.js:24:26)\n    at async Object.readProjectManifestOnly (/usr/local/Cellar/pnpm/4.7.1/libexec/lib/node_modules/pnpm/lib/node_modules/@pnpm/cli-utils/lib/readProjectManifest.js:12:22)\n    at async /usr/local/Cellar/pnpm/4.7.1/libexec/lib/node_modules/pnpm/lib/node_modules/@pnpm/plugin-commands-installation/lib/link.js:115:30"
    }
  }
```

### Expected behavior:

Installing Angular-CLI, or any other command-line tool, works.

### Actual behavior:

The .bin files are not linked to any bin directory.
Linking manually does not work because of a wrong path.

### Probable cause:

Interestingly, the error mentions path `/usr/local/Cellar/node/13.7.0/pnpm-global/node_modules/@angular/cli`, but `/usr/local/Cellar/node/13.7.0/pnpm-global/node_modules` does not exist! There is only `/usr/local/Cellar/node/13.7.0/pnpm-global/`**`3/`**`node_modules`.

Passing the full path also prints the error, despite `package.json` clearly existing:

```sh
$ pnpm link -g '/usr/local/Cellar/node/13.7.0/pnpm-global/3/node_modules/@angular/cli'
 ERROR  No package.json (or package.yaml, or package.json5) was found in "/usr/local/Cellar/node/13.7.0/pnpm-global/node_modules/usr/local/Cellar/node/13.7.0/pnpm-global/3/node_modules/@angular/cli".
$ ls '/usr/local/Cellar/node/13.7.0/pnpm-global/3/node_modules/@angular/cli/package.json'
/usr/local/Cellar/node/13.7.0/pnpm-global/3/node_modules/@angular/cli/package.json
```

### Additional information:

 - `node -v` prints: v13.7.0
 - Windows, OS X, or Linux?: macOS 10.15.3

**Comment by Sceat** [comment]
Same problem here, it tries to install from 
```
.nvm/versions/node/x/pnpm-global/node_modules
```
but node_modules are actually inside
```
.nvm/versions/node/x/pnpm-global/3/node_modules
```
so as a workaround i just copied the content of `3/` to `../3`
```
./  ../  3/  node_modules/  package.json  pnpm-lock.yaml
```

**Comment by Clashsoft** [comment]
Did the same, but now `pnpm link -g @angular/cli` outputs the following error:

```
> @angular/cli@9.1.7 postinstall /usr/local/Cellar/node/14.3.0/pnpm-global/node_modules/@angular/cli/pnpm-global/3
> node ./bin/postinstall/script.js

internal/modules/cjs/loader.js:1032
  throw err;
  ^

Error: Cannot find module '/usr/local/Cellar/node/14.3.0/pnpm-global/3/node_modules/.pnpm/registry.npmjs.org/@angular/cli/9.1.7/node_modules/@angular/cli/pnpm-global/3/bin/postinstall/script.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1029:15)
    at Function.Module._load (internal/modules/cjs/loader.js:898:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
 ERROR  Command failed with exit code 1.
```

**Comment by yaodingyd** [comment]
Linking a globally installed package seems unnecessary. Npm would prevent you from doing that. 
I think the issue is more like why the global install doesn't link the bin in the first place.

can you try with the latest version? it's working for me: `pnpm i -g @angular/cli && ng` would work

**Comment by Clashsoft** [comment]
Yes, of course `install -g` should already do the linking. With the latest version however, I am running into #2649.

**Comment by Clashsoft** [comment]
This seems to be solved as of v5.2.8.
